client/cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -629,4 +629,69 @@ describe('Dashboard', () => {
       });
     });
   });
+
+  context('viewport width is at 800px', () => {
```

Comment:
With `context` how about move `cy.viewport(800, 800);` to its `beforeEach` hook?

You'll have to invert tests order so the assertion for one column mode comes first, but it seems to make sense:
```javascript
context('viewport width is at 800px', () => {
  beforeEach(function () {
    cy.visit(this.dashboardUrl);
    cy.viewport(800, 800);
  });

  it('shows widgets with full width', () => {
    cy.get('@textboxEl').should(($el) => {
      expect($el.width()).to.eq(785);
    });
    cy.viewport(801, 800);
    cy.get('@textboxEl').should(($el) => {
      expect($el.width()).to.eq(393);
    });
  });
});
```

Reply:
@gabrieldutra I went for this, wdyt?
```js
describe('Viewport width affects presentation', () => {
  context('viewport width is at 800px', () => {
    it('shows widgets with full width' ...
    it('hides edit option' ...
    it('disables edit mode' ...
  });

  context('viewport width is above 800px', () => {
    it('shows widgets with specified width' ...
    it('shows edit option' ...
    it('enables edit mode' ...
  });

  context('viewport width is at 767px', () => {
    it('hides menu button' ...
  });

  context('viewport width is above 767px', () => {
     it('shows menu button' ...
  });
});
```

Reply:
I went for sth leaner:

![Screen Shot 2019-04-07 at 18 48 11](https://user-images.githubusercontent.com/486954/55686071-b97c2280-5965-11e9-8918-e243e2db1b62.png)